File: TwoSum2.py
# https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
class Solution:
    # HashMap
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        map={}
        out=[]
        for i in range(len(numbers)):
            x=target-numbers[i]
            if x in map:
                return [map[x]+1,i+1]
            else: 
                map[numbers[i]]=i
        return out
    
    # Two Pointers
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l=0
        r=len(numbers)-1
        while l<r:
            if numbers[l]+numbers[r]==target:
                return [l+1,r+1]
            elif numbers[l]+numbers[r]>target:
                r-=1
            elif numbers[l]+numbers[r]<target:
                l+=1
         

File: TwoSum2.java
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
public class TwoSum2 {
    public int[] twoSum(int[] numbers, int target) {
        int l=0;
        int r=numbers.length-1;
        int[] res=new int[2];
        while(l<r){
            if(numbers[l]+numbers[r]<target){
                l++;
            }
            else if(numbers[l]+numbers[r]>target){
                r--;
            }
            else{
                res[0]=(l+1);
                res[1]=(r+1);
                return res;
            }
        }
        return res;
    }
}
